### DISCUSSION TITLE
how to setup phpactor to work with laravel ?

### DISCUSSION THREAD

**Discussion by noor-tg** [phpactor/phpactor#2741] [Q&A]
Hi, I use [Helix Editor](https://github.com/helix-editor/helix) . And I develop websites with laravel. I see that phpactor has some features that Intelephense don't. helix can run phpactor as lsp. but there is many false positives when start phpactor in laravel project.

So any guide on how to config phpactor to work with laravel correctly ?

for your information the project contain [Laravel Ide Helper](https://github.com/barryvdh/laravel-ide-helper) as dependancy.

**Comment by dantleech** [comment]
This is a common request, I'm not a Laravel developer so I've never had to deal with the issues personally.

From what I can gather the main issue is that although you can generate stubs with the Helper, the stubs _replace_ the real definitions and do not _augment_ them?

So for example I can add stubs:

```json
{
    "$schema": "/home/daniel/www/phpactor/phpactor/phpactor.schema.json",
    "language_server_phpstan.enabled": true,
    "indexer.stub_paths": [
        "/home/daniel/www/phpactor/phpactor/vendor/jetbrains/phpstorm-stubs",
        "/home/daniel/www/firefly/firefly-iii/stubs"
    ]
}
```

but as I understand it this means that the original definitions are ignored and the incomplete stubs are used instead.

maybe @fatboyxpc or @haringsrob can comment on the state of Phpactor vs. Laravel and suggest what might be the next most beneficial step?

**Reply by noor-tg** [reply]
I tried to config the stubs.
But when run `phpactor index:build`, there is error out about not finding the dir that contain the stubs. But I am sure it is there.

I use stubs from laravel-ide-helper

**Reply by dantleech** [reply]
You might need to move them to a stubs dir and add the dir as with my config above (we should fiz it so that you can specify a single file)

**Reply by dantleech** [reply]
Maybe needs to be an absolute path?

**Reply by noor-tg** [reply]
I used `%project_root%`

**Reply by noor-tg** [reply]
> Maybe needs to be an absolute path?

I will test using absolute path
